lib/ide/src/executor/web.rs

```diff
@@ -0,0 +1,104 @@
+//! Module defining `JsExecutor` - an executor that tries running until stalled
+//! on each animation frame callback call.
+
+use crate::prelude::*;
+
+use basegl::control::callback::CallbackHandle;
+use basegl::control::EventLoopCallback;
+use basegl::control::EventLoop;
+use futures::task::LocalSpawn;
+use futures::task::LocalFutureObj;
+use futures::task::SpawnError;
+use futures::executor::LocalPool;
+use futures::executor::LocalSpawner;
+
+/// Executor. Uses a single-threaded `LocalPool` underneath, relying on basegl's
+/// `EventLoop` to do as much progress as possible on every animation frame.
+#[derive(Debug)]
+pub struct JSExecutor {
```

Comment:
It is named `JSExecutor` and the docs do not tell anything about JS connection here. They tell it is just executor. What is the difference between JSExecutor and Executor then ?

Reply:
There is no difference. There is no such thing as the "Executor". Executor is a concept. JSExecutor is an executor.

But I feel sentiment that actually this executor is not directly about JS. The event loop is about JS, because it implements on top of `Window.requestAnimationFrame` but that's implementation detail of a different module.
Will rename.